src/Microsoft.Diagnostics.NETCore.Client/ReversedServer/ReversedDiagnosticsServer.cs

```diff
@@ -82,91 +88,63 @@ public void Dispose()
-        /// <remarks>
-        /// This will only provide endpoint information on the first time a runtime connects to the server.
-        /// If a connection is removed using <see cref="RemoveConnection(Guid)"/> and the same runtime instance,
-        /// reconnects after this call, then a new <see cref="IpcEndpointInfo"/> will be produced.
-        /// </remarks>
-        public async Task<IpcEndpointInfo> AcceptAsync(CancellationToken token)
+        public void Start()
+        {
+            Start(MaxAllowedConnections);
+        }
+
+        /// <summary>
+        /// Starts listening at the transport path for new connections.
+        /// </summary>
+        /// <param name="maxConnections">The maximum number of connections the server will support.</param>
+        public void Start(int maxConnections)
         {
             VerifyNotDisposed();
 
-            while (true)
+            if (IsStarted)
             {
-                Stream stream = null;
-                IpcAdvertise advertise = null;
-                try
-                {
-                    stream = await _transport.AcceptAsync(token).ConfigureAwait(false);
-                }
-                catch (Exception ex) when (!(ex is OperationCanceledException))
-                {
-                    // The advertise data could be incomplete if the runtime shuts down before completely writing
-                    // the information. Catch the exception and continue waiting for a new connection.
-                }
+                throw new InvalidOperationException(nameof(ReversedDiagnosticsServer.Start) + " method can only be called once.");
+            }
 
-                token.ThrowIfCancellationRequested();
+            _listenTask = ListenAsync(maxConnections, _disposalSource.Token);
+        }
 
-                if (null != stream)
-                {
-                    // Cancel parsing of advertise data after timeout period to
-                    // mitigate runtimes that write partial data and do not close the stream (avoid waiting forever).
-                    using var parseCancellationSource = new CancellationTokenSource();
-                    using var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(token, parseCancellationSource.Token);
-                    try
-                    {
-                        parseCancellationSource.CancelAfter(ParseAdvertiseTimeout);
+        /// <summary>
+        /// Gets endpoint information when a new runtime instance connects to the server.
+        /// </summary>
+        /// <param name="token">The token to monitor for cancellation requests.</param>
+        /// <returns>A task that completes with a <see cref="IpcEndpointInfo"/> value that contains information about the new runtime instance connection.</returns>
+        public async Task<IpcEndpointInfo> AcceptAsync(CancellationToken token)
```

Comment:
I think @wiktork already mentioned this elsewhere, but it would be useful to have a synchrnous version of these APIs.  Additionally, it would be useful to have an overload of the async APIs that doesn't require a `CancellationToken`, similar to the `System.Net.Socket.AcceptAsync` API.

Reply:
Regarding additional APIs, please see https://github.com/dotnet/diagnostics/pull/1417#discussion_r468186387

Doing those changes now will make this PR very complicated to review. This PR is only meant to refactor the existing implementation so that listening is separate from acquiring endpoints.

IMO, the `System.Net.Socket` API surface is not a good representation of what an async API surface should look like (especially since `Socket` doesn't have ANY Task-based async representation). Also, adding overloads for not having to specify the CancellationToken doesn't seem very helpful or is easily specified by the consumer:

`var endpointInfo = server.AcceptAsync(CancellationToken.None);`

I would say, at best, we can specify a default value of the token parameter so that it's not needed to be specified:

`Task<IpcEndpointInfo> AcceptAsync(CancellationToken token = default)`

But I don't know if we want to encourage the consumer to not specify the CancellationToken. So I'm inclined to leave as-is since the barrier to usage should be very low on these simple APIs.